web/src/model/topology.ts

```diff
@@ -37,95 +39,7 @@ export enum LayoutName {
   colaGroups = 'ColaGroups'
 }
 
-export enum TopologyGroupTypes {
-  none = 'none',
-  clusters = 'clusters',
-  clustersZones = 'clusters+zones',
-  clustersHosts = 'clusters+hosts',
-  clustersNamespaces = 'clusters+namespaces',
-  clustersOwners = 'clusters+owners',
-  zones = 'zones',
-  zonesHosts = 'zones+hosts',
-  zonesNamespaces = 'zones+namespaces',
-  zonesOwners = 'zones+owners',
-  hosts = 'hosts',
-  hostsNamespaces = 'hosts+namespaces',
-  hostsOwners = 'hosts+owners',
-  namespaces = 'namespaces',
-  namespacesOwners = 'namespaces+owners',
-  owners = 'owners'
-}
-
-export const getGroupsForScope = (scope: MetricScopeOptions) => {
-  switch (scope) {
-    case MetricScopeOptions.CLUSTER:
-      return [TopologyGroupTypes.none];
-    case MetricScopeOptions.ZONE:
-      return [TopologyGroupTypes.none, TopologyGroupTypes.clusters];
-    case MetricScopeOptions.HOST:
-      return [TopologyGroupTypes.none, TopologyGroupTypes.clusters, TopologyGroupTypes.zones];
-    case MetricScopeOptions.NAMESPACE:
-      return [
-        TopologyGroupTypes.none,
-        TopologyGroupTypes.clusters,
-        TopologyGroupTypes.clustersHosts,
-        TopologyGroupTypes.zones,
-        TopologyGroupTypes.zonesHosts,
-        TopologyGroupTypes.hosts
-      ];
-    case MetricScopeOptions.OWNER:
-      return [
-        TopologyGroupTypes.none,
-        TopologyGroupTypes.clusters,
-        TopologyGroupTypes.clustersZones,
-        TopologyGroupTypes.zones,
-        TopologyGroupTypes.zonesHosts,
-        TopologyGroupTypes.zonesNamespaces,
-        TopologyGroupTypes.zonesOwners,
-        TopologyGroupTypes.hosts,
-        TopologyGroupTypes.hostsNamespaces,
-        TopologyGroupTypes.namespaces
-      ];
-    case MetricScopeOptions.RESOURCE:
-    default:
-      return Object.values(TopologyGroupTypes);
-  }
-};
-
-export const isGroupEnabled = (group: TopologyGroupTypes, enabledScopes: FlowScope[]): boolean => {
-  return (
-    (enabledScopes.includes('cluster') || !group.includes(TopologyGroupTypes.clusters)) &&
-    (enabledScopes.includes('zone') || !group.includes(TopologyGroupTypes.zones)) &&
-    (enabledScopes.includes('host') || !group.includes(TopologyGroupTypes.hosts)) &&
-    (enabledScopes.includes('namespace') || !group.includes(TopologyGroupTypes.namespaces)) &&
-    (enabledScopes.includes('owner') || !group.includes(TopologyGroupTypes.owners))
-  );
-};
-
-export const getStepIntoNext = (current: FlowScope, allowedScopes: FlowScope[]): FlowScope | undefined => {
```

Comment:
I think you need to reintroduce this function (with a few changes), because it was a recursive function that would continue looking for "next" step in case the current found one is not allowed.
This allows to "jump" over disabled scopes. Two examples:
- When you enable multi-clusters but not zones: stepping into cluster should bring you directly to nodes, by-passing zones
- When disabling Loki, stepping into a Owner should not bring you to Resource (it should be disallowed)

The current implementation breaks these use cases

Reply:
Restored recursive behavior + added tests
https://github.com/netobserv/network-observability-console-plugin/pull/634/commits/00e7f3b36caa67bff3b7fe6eff19ad100de45829

Reply:
thanks for the tests, indeed that was missing